spec/models/due_date_spec.rb

```
@@ -184,6 +184,79 @@
+    end
+  end
+
+  describe '.deadline_open_for?' do
+    let(:assignment) { Assignment.create!(id: 3001, name: 'Window Assignment', instructor:) }
+
+    it 'returns true when there is no assignment context' do
+      expect(DueDate.deadline_open_for?(action: :submission, assignment: nil)).to be true
+    end
+
+    it 'returns true when no upcoming due date exists' do
+      allow(DueDate).to receive(:next_due_date_for).with(action: :submission, assignment: assignment, topic: nil).and_return(nil)
```

RuboCop (1.86.1)

[convention] Line is too long. [129/120]

(Layout/LineLength)